src/server/rdb_save.cc

```diff
@@ -1278,26 +1296,57 @@ error_code RdbSaver::Impl::WriteRecord(io::Bytes src) {
   return ec;
 }
 
-void RdbSaver::Impl::StartSnapshotting(bool stream_journal, const Cancellation* cll,
-                                       EngineShard* shard) {
+void RdbSaver::Impl::PushSnapshotData(Context* cntx, string record) {
+  if (cntx->IsCancelled()) {
+    return;
+  }
+  if (save_mode_ == SaveMode::RDB) {  // Rdb write to channle
```

Comment:
you use `save_mode_ == SaveMode::RDB` when you create the channel. here you can just check `if (channel_)` to localize business logic into a single place in the code.